### ISSUE TITLE
Incorrect incorrect arity warning for symbol with meta in function call

### ISSUE THREAD

**Issue by mpietrzak** [cursiveclojure/cursive#782]
Cursive incorrectly counts symbol with meta as separate arguments in function call, resulting in arity warning.

Sample code:

```clojure
(ns cursari.core
  (:gen-class)
  (:import (clojure.lang IObj)))


(defn fn3
  [^IObj x y z]
  (println (format "x: %s (%s), y: %s, z: %s" x (meta x) y z)))


(defn -main
  "I don't do a whole lot ... yet."
  []
  (let [c1 (fn3 '^ {:foo 1 :bar 2} s1 2 3) ; incorrect arity warning
        c2 (fn3 (with-meta 's2 {:foo 1 :bar 2}) 2 3)]
    (println c1)
    (println c2)))
```

Screenshot:

![zrzut ekranu 2015-03-09 o 13 09 31](https://cloud.githubusercontent.com/assets/2089393/6554584/acc51320-c65d-11e4-81cd-d5674c981f3d.png)

**Comment by cursiveclojure** [comment]
The problem here is that the metadata syntax is incorrect. Changing the invocation to `(fn3 ^{:foo 1 :bar 2} s1 2 3)` works as expected.

**Comment by mpietrzak** [comment]
After changing invocation like you say it does not work at all, but I guess this is odd syntax and explicit with-meta is cleaner anyway, so maybe there's no point to support that (even if it is correct syntax).

Thanks & best regards

**Comment by cursiveclojure** [comment]
I actually didn't think the syntax you had originally was correct - I may have been wrong. I'll investigate this - if that syntax is correct Cursive should definitely support it.

**Comment by cursiveclojure** [comment]
My apologies, you were absolutely correct. I'll investigate why that syntax works because I don't fully understand it but I'll add support for it.

**Comment by mpietrzak** [comment]
No worries.
I'm not sure if it's correct either, but I'm pretty sure Clojure accepts that. Might as well be obscure bug in Clojure…
Thanks for looking into that.

**Comment by cursiveclojure** [comment]
No, I don't think it's an obscure bug, but it depends on some pretty deep knowledge of the reader (I think the quote quotes the following symbol, but the metadata has to be inside the quote to apply the meta to the same symbol). It's definitely an edge case though.

**Comment by imrekoszo** [comment]
Just stumbled upon another repro case for this in [clojure koans](https://github.com/functional-koans/clojure-koans/blob/master/src/koans/23_meta.clj#L13)

![image](https://user-images.githubusercontent.com/1116295/186485919-f0ed5261-1cd4-49e9-9284-b436b9c345be.png)

- Cursive thinks `meta` is called with 2 args
- And it also doesn't apply the quote to the symbol after the meta map so `Giants` is marked as cannot be resolved
